### PR TITLE
Improve Swagger typings for getAll

### PR DESCRIPTION
**Pull request by azizdiakite**
## Issue I worked on

see https://issues.openmrs.org/browse/RESTWS-

## Checklist: I completed these to help reviewers :)

- [ ] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ ] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ ] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java [hunk 1]
```diff
@@ -132,7 +132,7 @@ private void BuildJSON() {
 			try {
 				initSwagger();
 				addPaths();
-				addDefaultDefinitions();
+				//addDefaultDefinitions();
```
Comment (ibacher) [inline omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java:135]: Is there a reason we don't want this? And if so, can we just delete the method?

File: omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java [hunk 2]
```diff
@@ -1069,7 +1069,10 @@ private Operation createOperation(DelegatingResourceHandler<?> resourceHandler,
 			
 			operation.setSummary("Fetch all non-retired");
 			operation.setOperationId("getAll" + getOperationTitle(resourceHandler, true));
-			operation.addResponse("200", response200.schema(new RefProperty("#/definitions/FetchAll")));
+			operation.addResponse("200",
+					response200.schema(new ArrayProperty(
+						new RefProperty(getSchemaRef(resourceName, resourceParentName, OperationEnum.get)))));
```
Comment (ibacher) [inline omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java:1074]: I'm not deeply familiar with Swagger. Could you leave a comment with a brief description of how this changes things? Is it replacing the definition of the return value with something that matches the `GET` request?

File: omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java (hunk 1 above)
Comment (azizdiakite) [reply omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java:135]: yes sure

File: omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java (hunk 2 above)
Comment (chibongho) [reply omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java:1073]: Currently, the `getAll` operation on a resouce is not properly typed in swagger. It returns a generic type called `FetchAll`, which has no typing information of the actual array being returned. (For example, `getAll` on patients should return `PatientGet[]`). This change should fix that.

Here's the typing for `getAll` on Patient currently:
image

**Review by ibacher — APPROVED** [review]

**Pull request merged**